tests/test_camera_readout.py

```diff
@@ -5,57 +5,91 @@
+        self.eimage_file = os.path.join('data', 'lsst_e_161899_R22_S11_r.fits.gz')
+        self.image = galsim.fits.read(self.eimage_file)
+        # Also, this file was made in phosim convention, which swaps x,y relative
+        # to the normal convention.  So we need to transpose the image after reading it.
+        self.image = transpose(self.image)
+        self.config = {
+            'image': {'random_seed': 1234},
+            'input': {
+                'opsim_meta_dict': {'file_name': 'tiny_instcat.txt'},
+            },
+            'output': {
+                'readout' : {
+                    'readout_time': 3,
+                    'dark_current': 0.02,
+                    'bias_level': 1000.,
+                    'pcti': 1.e-6,
+                    'scti': 1.e-6,
+                }
+            },
+            'index_key': 'image_num',
+            'image_num': 0,
+            'det_name': 'R20_S00',
+            'exp_time': 30,
+        }
+
+        galsim.config.SetupConfigRNG(self.config)
+        self.logger = galsim.config.LoggerWrapper(None)
+        self.readout = imsim.CameraReadout()
+        self.readout_config = self.config['output']['readout']
+        galsim.config.ProcessInput(self.config)
+        self.readout.initialize(None,None, self.readout_config, self.config, self.logger)
+        self.readout.ensureFinalized(self.readout_config, self.config, [self.image], self.logger)
 
     def tearDown(self):
         pass
 
     def test_create_from_eimage(self):
         "Test the .create_from_eimage static method."
-        self.assertAlmostEqual(self.image_source.exptime, 30.)
-        self.assertTupleEqual(self.image_source.eimage_data.shape, (4072, 4000))
-        self.assertTupleEqual(
-            self.image_source.amp_images['R22_S11_C00'].getArray().shape,
-            (2048, 544))
+        hdus = self.readout.final_data
+        self.assertAlmostEqual(hdus[0].header['EXPTIME'], 30.)
+        self.assertTupleEqual(self.image.array.shape, (4000, 4072))
+        for i in range(1,16):
+            self.assertTupleEqual(hdus[i].data.shape, (2048, 544))
 
     def test_get_amplifier_hdu(self):
         "Test the .get_amplifier_hdu method."
-        hdu = self.image_source.get_amplifier_hdu('R22_S11_C10')
+        hdus = self.readout.final_data
+        hdu = hdus[1]
+        self.assertEqual(hdu.header['EXTNAME'], "Segment10")
         self.assertEqual(hdu.header['DATASEC'], "[4:512,1:2000]")
+        # XXX: This test is currently failing.  This DETSEC now seems to correspond to
+        #      hdu 17, which is C00, not C10.  C10 has DETSEC = [509:1,4000:2001]
+        #      I don't know what should be changed to fix this.  Is it an error in the code?
+        #      Or is the test wrong now after some change to the camera layout?
+        #      Or did I screw something up in trying to adapt the test from the old code?
         self.assertEqual(hdu.header['DETSEC'], "[509:1,1:2000]")
```

Comment:
There's a bit of history regarding this test, but the bottom line is that I think the `DETSEC` value should be changed to `"[509:1,4000:2001]"` here for C10 and `"[3563:3055,4000:2001]"` for C17 below.  If we were to test e2V CCDs, those would be `"[512:1,4004:2003]"` and `"[3584:3073,4004:2003]"`, respectively.

The back story on this test is that the current `DETSEC` values are based on Camera document [LCA-10140](https://ls.st/lca-10140), but more recently, the Camera team decided to adopt the DM conventions for displaying mosaics of the focal plane.   Regarding amp orientation and location, the DM conventions amount to rotating images like those seen on sheets 1 or 5 of [LCA-13381](https://docushare.lsst.org/docushare/dsweb/Get/LCA-13381) 90 degrees clockwise.

Reply:
OK.  Thanks for tracking it down.  I fixed the test to match the values the code is giving.

Reply:
Jim, does 

"more recently, the Camera team decided to adopt the DM conventions for displaying mosaics of the focal plane. "

this only refer to this situation? Or will the camera team move towards DM conventions more generally?

Reply:
The Camera team is also moving towards using the Gen3 infrastructure and pipe_task framework for implementing new electro-optical analysis code, but that a much more involved transition and we may always rely to some extent on non-DM compliant scripts or code. For straight-forward things like the focal plane layout conventions, we've generally agreed that using DM conventions is best path forward, but I can't say definitively for any specific item.